src/Components/Components/src/Auth/IHostEnvironmentAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// An interface implemented by <see cref="AuthenticationStateProvider"/> classes that can receive authentication
+    /// state information from the host environment.
+    /// </summary>
+    public interface IHostEnvironmentAuthenticationStateProvider
+    {
+        /// <summary>
+        /// Supplies updated authentication state data to the <see cref="AuthenticationStateProvider"/>.
+        /// </summary>
+        /// <param name="authenticationStateTask">A task that resolves with the updated <see cref="AuthenticationState"/>.</param>
+        void SetAuthenticationState(Task<AuthenticationState> authenticationStateTask);
+    }
```

Comment:
I'm not super sure what this is needed, but the name immediately threw me off path, its too similar to IHostingEnvironment.

Reply:
Could we instead, have an HttpContextAuthenticationStateProvider in MVC, and have the FixedAuthenticationStateProvider contain this extra method that you can call during reconnection?

That way we don't have to carry around an additional interface.

Reply:
That doesn't work, because `StaticComponentRenderer` still has to initialize whatever `AuthenticationStateProvider` you have in DI. If the one you have there is a `FixedAuthenticationStateProvider`, it wouldn't be able to invoke any methods on it, because it lives in an assembly not referenced from ViewFeatures.

Reply:
What do you think about adding it to the base class instead?

Reply:
It's not applicable in general because, say, in client-side Blazor there's no host environment that can tell you what the authentication state is. The `AuthenticationStateProviderBase` is about *providing* the authentication state, not receiving it from someone else.

I don't mind renaming `IHostEnvironmentAuthenticationStateProvider` to something else if you prefer.

Reply:
Alternatively we could make ViewFeatures reference Components.Server and do all this on the concrete type for FixedAuthenticationStateProvider. 

Can’t remember why we didn’t want ViewFeatures to reference Components.Server originally. Can you?

Reply:
Yeah, but this is not about the interface name, its about the functionality. the ABC can simply have a
`SetAuthenticationState` you can use to set the authstate and that way you don't have to implement your own.

You can make it virtual so that if your implementation doesn't support setting it multiple times you throw, but in general I think this is similar to HttpContext.User, we provide a default mutable implementation, and if you choose to modify it is up to you.

I would not guard against users trying to change the auth state through the type system.

Reply:
We don't want MVC referencing signalr.

Reply:
> We don't want MVC referencing signalr.

Ah yes, now I remember.

Reply:
> Yeah, but this is not about the interface name, its about the functionality. the ABC can simply have a
SetAuthenticationState you can use to set the authstate and that way you don't have to implement your own.

It is about the functionality, yes. Setting the state has no place on the base contract as it’s not a concept that exists generally (at all, not just multiple times). That’s why I’m avoiding putting it there.

Reply:
@javiercn Do you have any further concerns about this?

Reply:
No, I think we have a different trade-off balance, so I leave it to your better judgement.